daq/gateway.py

```diff
@@ -216,19 +187,16 @@ def get_targets(self):
 
     def terminate(self):
         """Terminate this instance"""
-        assert not self.targets, 'gw %s has targets %s' % (self.name, self.targets)
+        assert not self.targets, 'gw %s has targets %s' % (
+            self.name, self.targets)
         LOGGER.info('Terminating gateway %d/%s', self.port_set, self.name)
-        if self.dhcp_monitor:
-            self.dhcp_monitor.cleanup()
-        self._scan_finalize()
-        if self.host:
+        for _, dhcp_server in self._dhcp_servers.items():
             try:
-                self.host.terminate()
-                self.runner.remove_host(self.host)
-                self.host = None
+                dhcp_server.terminate()
```

Comment:
looks like you lost the runner.remove_host

Reply:
that's in dhcp_server's terminate